askama_derive/src/config.rs

```diff
@@ -189,35 +206,68 @@ impl Config {
         start_at: Option<&Path>,
         file_info: Option<FileInfo<'_>>,
     ) -> Result<Arc<Path>, CompileError> {
-        let path = 'find_path: {
-            if let Some(root) = start_at {
-                let relative = root.with_file_name(path);
-                if relative.exists() {
-                    break 'find_path relative;
-                }
-            }
-            for dir in &self.dirs {
-                let rooted = dir.join(path);
-                if rooted.exists() {
-                    break 'find_path rooted;
-                }
-            }
-            return Err(CompileError::new(
-                format_args!(
+        let path = Path::new(path);
+        let err = match find_template_sub(path, &self.dirs, start_at, self.caller_dir())
+            .map(|p| p.canonicalize())
+        {
+            Some(Ok(path)) => return Ok(path.into()),
+            Some(Err(err)) => Some(err),
+            None => None,
+        };
+
+        Err(CompileError::new(
+            match err {
+                Some(err) => fmt_left!(
+                    move "could not canonicalize path {:?}: {err}",
+                    path.display(),
+                ),
+                None => fmt_right!(
                     "template {:?} not found in directories {:?}",
-                    path, self.dirs,
+                    path.display(),
+                    self.dirs,
```

Comment:
Is this accurate? We check more directories than just `self.dirs` (even before the changes in the PR). I believe this should be changed so that this displays `self.dirs` along with `start_at` and `caller_dir()`.